django/db/backends/mysql/creation.py

```diff
@@ -17,24 +17,42 @@ def sql_table_creation_suffix(self):
             suffix.append('COLLATE %s' % test_settings['COLLATION'])
         return ' '.join(suffix)
 
+    def _execute_create_test_db(self, cursor, parameters, keepdb=False):
+        try:
+            if keepdb:
+                # If the database should be kept, add "IF NOT EXISTS" to avoid
+                # "database exists" error, also temporarily disable "database
+                # exists" warning.
+                cursor.execute('''
+                    SET sql_notes = 0;
+                    CREATE DATABASE IF NOT EXISTS %(dbname)s %(suffix)s;
+                    SET sql_notes = 1;
```

Comment:
what if it was set to 0 by connection init or server param? We should just reset it back to the original value...

```sql
SET @_tmp_sql_notes := @@sql_notes;
SET sql_notes = 0;
CREATE DATABASE IF NOT EXISTS ...
SET sql_notes = @_tmp_sql_notes;
```